### PR TITLE
Merging `integrations` plugin.

### PR DESCRIPTION
**Pull request by dennisoelkers**
# Note: When merging this PR, make sure NOT to squash it in order to keep the commit history!

## Motivation and Context

Refs Graylog2/graylog-plugin-enterprise#3558.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by bernd — COMMENTED** [review]
@dennisoelkers Let's put the Java code into the `graylog2-server/src/main/java/org/graylog/plugins` directory like we did for the recent threatintel plugin merge and where the other merged plugins are.

```
$ ls -l graylog2-server/src/main/java/org/graylog/plugins/
total 32
drwxr-xr-x  2 bernd bernd 4096 Jun 21 15:51 beats
drwxr-xr-x  6 bernd bernd 4096 Jun 21 15:51 cef
drwxr-xr-x  4 bernd bernd 4096 Jun 21 15:51 map
drwxr-xr-x  9 bernd bernd 4096 Jun  9  2021 netflow
drwxr-xr-x 13 bernd bernd 4096 Jun 21 15:51 pipelineprocessor
drwxr-xr-x 13 bernd bernd 4096 Jul 31 08:26 sidecar
drwxr-xr-x  7 bernd bernd 4096 Jun 21 15:51 threatintel
drwxr-xr-x  8 bernd bernd 4096 Jun 28 12:19 views
```

**Comment by dennisoelkers** [comment]
@bernd: I thought about that too, but unfortunately parts of our paths are also bound to the package name of the plugin, so I refrained from moving it in this PR to avoid introducing regressions. I would still like to do it, but in a follow-up PR so I can focus on it.

**Comment by bernd** [comment]
> @bernd: I thought about that too, but unfortunately parts of our [REST api paths](https://github.com/Graylog2/graylog2-server/pull/16274/files#diff-def568a64c08113b87488f427366afc1e6b0db85054c98a7ac0f77c604efc907R30-R43) are also bound to the package name of the plugin, so I refrained from moving it in this PR to avoid introducing regressions. I would still like to do it, but in a follow-up PR so I can focus on it.

@dennisoelkers Ah, right. I missed that. Thanks!

File: graylog2-server/src/main/java/org/graylog/integrations/Integrations.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.integrations;
+
+/**
+ * This is the plugin. Your class should implement one of the existing plugin
+ * interfaces. (i.e. AlarmCallback, MessageInput, MessageOutput)
+ */
+public class Integrations {
```
Comment (bernd) [inline graylog2-server/src/main/java/org/graylog/integrations/Integrations.java:23]: This file seems to be unused.

**Review by bernd — CHANGES_REQUESTED** [review]
@dennisoelkers Let's add a `package-info.java` file with the note about the package name like we did for the AWS plugin: https://github.com/Graylog2/graylog2-server/blob/master/graylog2-server/src/main/java/org/graylog/aws/package-info.java

File: graylog2-server/src/main/java/org/graylog/integrations/Integrations.java (hunk 1 above)
Comment (dennisoelkers) [reply graylog2-server/src/main/java/org/graylog/integrations/Integrations.java:23]: :white_check_mark:

**Comment by dennisoelkers** [comment]
@bernd: Thanks for the hint! Added the `package-info.java`.

**Review by danotorrey — APPROVED** [review]
@dennisoelkers This LGTM and tests successfully. All the code is migrated, and the package names line up. Well done!

I successfully tested creating an AWS Kinesis/Cloudwatch input, the Slack/Teams notification, and the DNS Adapter pre-upgrade (without this PR checked-out). All work successfully after upgrade, and also can be edited/created after successfully too.

**Review by kingzacko1 — APPROVED** [review]

**Review by bernd — APPROVED** [review]
My comments got addressed. I only looked at some code infrastructure changes and did not test the plugin.

Thanks, @dennisoelkers! :pray:

**Pull request merged**